src/scanner.rs

```diff
@@ -2486,7 +2501,10 @@ impl<T: Iterator<Item = char>> Scanner<T> {
         if self.simple_key_allowed {
             let required = self.flow_level == 0
                 && self.indent == (self.mark.col as isize)
-                && self.indents.last().unwrap().needs_block_end;
```

Comment:
Removing this unwrap would silence a potential error in the scanner logic.

Reply:
This could also be `self.indents.last().map_or(false, |ident| ident.needs_block_end)`.